plat/linuxu/memory.c

```diff
@@ -127,10 +127,13 @@ static int __linuxu_plat_initrd_init(void)
 
 int ukplat_memregion_count(void)
 {
+	static int init = 0;
 	static int have_heap = 0;
 	static int have_initrd = 0;
```

Comment:
Also, I believe that my example from the previously closed PR dropped the `static` keyword from `have_heap` and `have_initrd`. Why do you think they are still needed? As far as I can tell, `static` for `init` would be enough.

Reply:
Why do we keep these variables? I think it would be a cleaner if we use only the rc and remove both `have_heap` and `have_initrd`

Reply:
Yeah, agreed, you are right. Something like this
```c
int ukplat_memregion_count(void)
{
	static int init;

	if (init)
		return init;
		
	init = !__linuxu_plat_heap_init() + !__linuxu_plat_initrd_init();

	return init;
}
```
could work as well and you can drop everything but `init` :neckbeard:.
Or 
```c
int ukplat_memregion_count(void)
{
	static int init;
	int rc;

	if (init)
		return init;
	
	rc = __linuxu_plat_heap_init();
	init += (rc == 0) ? 1 : 0 ;

	rc = __linuxu_plat_initrd_init();
	init += (rc == 0) ? 1 : 0 ;
	
	return init;
}
```
and the list could go on :laughing: .
I don't think it matters that much in the end, though. @Daparrag choose whichever you like,  as long as it works, looks decent and does not have any unnecessary statements (I would personally go with the latter).

Reply:
Good Thank you so much for your review and I will start working on that.